field_encrypt.module

```diff
@@ -117,33 +123,123 @@ function field_encrypt_form_field_add_form_builder($entity_type, \Drupal\field\E
   }
 
   if ($original_encryption !== $field_storage_config->getThirdPartySettings('field_encrypt')) {
-    // We need to process the field to either encrypt or decrypt the stored
-    // fields if the setting was changed.
-    $field_name = $field_storage_config->get('field_name');
-    $field_entity_type = $field_storage_config->get('entity_type');
-
-    /* @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities */
-    // @TODO: refactor logic for re-encrypting existing fields - currently broken, so commented out.
-    //    $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
-    //    if ($form_encryption === 1) {
-    //      $field_encrypt_process_entities->encryptStoredField($field_entity_type, $field_name);
-    //    }
-    //    elseif ($form_encryption === 0) {
-    //      $field_encrypt_process_entities->decryptStoredField($field_entity_type, $field_name);
-    //    }
+    // Only re-encrypt field data if this field contains actual data.
+    if ($field_storage_config->hasData()) {
+      // We need to process the field to either encrypt or decrypt the stored
+      // fields if the setting was changed.
+      $storage = $form_state->getStorage();
+      $storage['field_encrypt_update'] = TRUE;
+      $storage['field_encrypt_field_name'] = $field_storage_config->get('field_name');
+      $storage['field_encrypt_entity_type'] = $field_storage_config->get('entity_type');
+      $storage['original_encryption_settings'] = $original_encryption;
+      $form_state->setStorage($storage);
+    }
+  }
+}
+
+/**
+ * Submit handler to update encryption after saving field storage settings.
+ *
+ * @param $form
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ */
+function field_encrypt_update_encryption_form_submit(&$form, FormStateInterface $form_state) {
+  $storage = $form_state->getStorage();
+  $entity_type_manager = \Drupal::service('entity_type.manager');
+  if (isset($storage['field_encrypt_update']) && $storage['field_encrypt_update'] == TRUE) {
```

Comment:
Can you add some more inline comments here? Especially around batch API operations